cypress/component/integration/Dropdown.cy.tsx

```diff
@@ -0,0 +1,415 @@
+              { icon: <MassaLogo size={32} />, item: 'account 5 name' },
+              { icon: <MassaLogo size={32} />, item: 'account 6 name' },
+            ]}
+          />,
+        );
+      });
+
+      it('should render', () => {
+        cy.get('[data-testid="dropdown"]').should('exist');
+      });
+
+      it('should match snapshot', () => {
+        compareSnapshot(cy, 'default-md-dropdown');
+      });
+    });
+
+    describe('default dropdown flow', () => {
+      beforeEach(() => {
+        cy.mount(
+          <Dropdown
+            options={[
+              {
+                icon: (
+                  <MassaToken
+                    data-testid="dropdown-option-1"
+                    size={32}
+                    className="bg-c-default rounded-full"
+                  />
+                ),
+                item: 'account 1 name',
+                onClick: () => console.log('option 1'),
+              },
+              {
+                icon: <MassaLogo data-testid="dropdown-option-2" size={32} />,
+                item: 'account 2 name with a biiiiig content',
+              },
+              {
+                icon: <MassaLogo data-testid="dropdown-option-3" size={32} />,
+                item: 'account 3 name',
+              },
+              {
+                icon: <MassaLogo data-testid="dropdown-option-4" size={32} />,
+                item: 'account 4 name',
+              },
+              {
+                icon: <MassaLogo data-testid="dropdown-option-5" size={32} />,
+                item: 'account 5 name',
+              },
+              {
+                icon: <MassaLogo data-testid="dropdown-option-6" size={32} />,
+                item: 'account 6 name',
+              },
+            ]}
+          />,
+        );
+      });
+
+      it('should render', () => {
+        cy.get('[data-testid="dropdown"]').should('exist');
+        it('should open and close when clicked', () => {
+          cy.get('[data-testid="dropdown"]').should('exist');
+
+          for (let i = 1; i <= 6; i++) {
+            cy.get(`[data-testid="dropdown-option-${i}"]`).should('exist');
+          }
+
+          cy.get('[data-testid="dropdown-option-1"]').should('be.visible');
+          for (let i = 2; i <= 6; i++) {
+            cy.get(`[data-testid="dropdown-option-${i}"]`).should(
+              'not.be.visible',
+            );
+          }
+        });
+
+        // open
+        cy.get('[data-testid="dropdown-button"]').should('exist').click();
+
+        for (let i = 1; i <= 6; i++) {
+          cy.get(`[data-testid="dropdown-option-${i}"]`).should('be.visible');
+        }
+
+        // close
+        cy.get('[data-testid="dropdown-button"]').should('exist').click();
+
+        cy.get('[data-testid="dropdown-option-1"]').should('be.visible');
+        for (let i = 2; i <= 6; i++) {
+          cy.get(`[data-testid="dropdown-option-${i}"]`).should(
+            'not.be.visible',
+          );
+        }
```

Comment:
No need to check each element visibility. Check if the div wrapping it is visible or not. It should be enough! In the worst case test only one element is fine.